src/components/Checkbox.js

```diff
@@ -82,6 +82,13 @@ class Checkbox extends React.Component {
             return;
         }
 
+        const wasChecked = this.props.isChecked;
+
+        // If checkbox is checked and focused, make sure it's unfocused when pressed.
+        if (this.state.isFocused && wasChecked) {
+            this.onBlur();
+        }
```

Comment:
This caused a regression https://github.com/Expensify/App/issues/18801. The `onBlur` should not be called explicitly, it should be called only when indeed the checkbox is blurred. This PR is trying to achieve `focus-visible` styles using `focus state` which won't work correctly due to the nature of how focus-visible style is maintained in browsers.